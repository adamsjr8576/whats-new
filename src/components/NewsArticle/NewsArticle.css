.news-article {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  width: 450px;
  margin: 10px;
  background-color: white;
  height: 500px;
  opacity: .8;
}

.news-article:hover {
  box-shadow: 6px 6px #3E3E3E;
  opacity: 1;
}

.article-img {
  width: 100%;
  height: 250px;
}

.article-text {
  margin: 0 20px;
}

.link {
  display: flex;
  align-items: center;
  justify-content: space-around;
  background-color: #757576;
  height: 50px;
  width: 100%;
}

.article-link {
  color: black;
  outline: none;
  style: none;
  cursor: pointer;
  font-size: 125%;
}

.arrow-link {
  height: 40px;
  width: 50px;
}
