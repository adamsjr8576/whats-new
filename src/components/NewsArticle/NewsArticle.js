import React from 'react';
import './NewsArticle.css';
import arrow from '../../images/arrow.png'

const NewsArticle = ({headline, img, description, url}) => {
  return (
    <article className="news-article">
      <img className="article-img" src={img} alt="article"/>
      <h2 className="article-text">{headline}</h2>
      <p className="article-text">{description}</p>
      <section className="link">
      <a href={url} className="article-link">Link To Article</a>
      <a href={url} className="article-link"><img src={arrow} alt="arrow icon" className="arrow-link"/></a>
      </section>
    </article>
  );
}

export default NewsArticle;
