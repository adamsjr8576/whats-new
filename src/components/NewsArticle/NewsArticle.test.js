import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {

  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<NewsArticle
      headline={"Hello world!!"}
      img={"https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp"}
      description={"I am saying hello to the world!"}
      url={"https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html"}
      />)

      expect(wrapper).toMatchSnapshot()
  });

});
