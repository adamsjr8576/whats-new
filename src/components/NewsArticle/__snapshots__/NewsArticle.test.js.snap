// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed in correctly 1`] = `
<article
  className="news-article"
>
  <img
    alt="article"
    className="article-img"
    src="https://static01.nyt.com/images/2019/09/27/multimedia/27xp-spiderman/27xp-spiderman-superJumbo.jpg?quality=90&auto=webp"
  />
  <h2
    className="article-text"
  >
    Hello world!!
  </h2>
  <p
    className="article-text"
  >
    I am saying hello to the world!
  </p>
  <section
    className="link"
  >
    <a
      className="article-link"
      href="https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html"
    >
      Link To Article
    </a>
    <a
      className="article-link"
      href="https://www.nytimes.com/2019/09/27/business/media/spiderman-mcu-sony-marvel.html"
    >
      <img
        alt="arrow icon"
        className="arrow-link"
        src="arrow.png"
      />
    </a>
  </section>
</article>
`;
